static/js/composebox_typeahead.js

```diff
@@ -315,9 +317,9 @@ exports.content_typeahead_selected = function (item) {
     if (this.completing === 'emoji') {
         // leading and trailing spaces are required for emoji, except if it begins a message.
         if (beginning.lastIndexOf(":") === 0 || beginning.charAt(beginning.lastIndexOf(":") - 1) === " ") {
-            beginning = beginning.replace(/:\S+$/, "") + ":" + item.emoji_name + ": ";
+            beginning = (beginning.substring(0, beginning.length - this.token.length-1)+ ":" + item.emoji_name + ": ");
         } else {
-            beginning = beginning.replace(/:\S+$/, "") + " :" + item.emoji_name + ": ";
+            beginning = (beginning.substring(0, beginning.length - this.token.length-1) + " :" + item.emoji_name + ": ");
```

Comment:
can you explain this change?  Also you should spell the subtraction ` - 1` to match the style of nearby code.